ross/shaft_element.py

```diff
@@ -1201,3 +1201,911 @@ def section(
+        Default is zero.
+    torque : float, optional
+        Torque moment;
+        Default is zero.
+    shear_effects : bool, optional
+        Determine if shear effects are taken into account;
+        Default is True.
+    rotary_inertia : bool, optional
+        Determine if rotary_inertia effects are taken into account;
+        Default is True.
+    gyroscopic : bool, optional
+        Determine if gyroscopic effects are taken into account;
+        Default is True.
+    tag : str, optional
+        Element tag;
+        Default is None.
+
+    Returns
+    -------
+    A 6 degrees of freedom shaft element, with available gyroscopic, shear and rotary inertia effects.
+
+    Attributes
+    ----------
+    Poisson : float
+        Poisson coefficient for the element.
+    A : float
+        Element section area at half length.
+    A_l : float
+        Element section area at left end.
+    A_r : float
+        Element section area at right end.
+    beam_cg : float
+        Element center of gravity local position.
+    axial_cg_pos : float
+        Element center of gravity global position.
+        This should be used only after the rotor is built.
+        Default is None.
+    Ie : float
+        Ie is the second moment of area of the cross section about
+        the neutral plane.
+    phi : float
+        Constant that is used according to :cite:`friswell2010dynamics` to
+        consider rotary inertia and shear effects. If these are not considered
+        :math:`\phi=0`.
+    kappa : float
+        Shear coefficient for the element, determined from :cite:`Hutchingson2001` formulation.
+    alpha : float
+        Proportional damping coefficient, associated to the element Mass matrix
+    beta : float
+        Proportional damping coefficient, associated to the element Stiffness matrix
+
+    References
+    ----------
+    .. bibliography:: ../../../docs/refs.bib
+
+    Examples
+    --------
+    >>> from ross.materials import steel
+    >>> Euler_Bernoulli_Element = ShaftElement6DoF(
+    ...                         material=steel, L=0.5, idl=0.05, odl=0.1,
+    ...                         idr=0.05, odr=0.15,
+    ...                         alpha=0.01, beta=100,
+    ...                         rotary_inertia=False,
+    ...                         shear_effects=False)
+    >>> Euler_Bernoulli_Element.phi
+    0
+    """
+
+    @check_units
+    def __init__(
+        self,
+        L,
+        idl,
+        odl,
+        idr,
+        odr,
+        material=None,
+        n=None,
+        axial_force=0,
+        torque=0,
+        shear_effects=True,
+        rotary_inertia=True,
+        gyroscopic=True,
+        alpha=0,
+        beta=0,
+        tag=None,
+    ):
+
+        if idr is None:
+            idr = idl
+        if odr is None:
+            odr = odl
+
+        # Changing units to only their magnitude
+        L = L.m
+        idl = idl.m
+        odl = odl.m
+        idr = idr.m
+        odr = odr.m
```

Comment:
Hi @verg1lio ! 
You will need to remove this lines of code since we have changed the `check_units` decorator on commit 19433d79048346697537c8481e798ab6f0fe8716
![image](https://user-images.githubusercontent.com/18506378/75560245-90ec9380-5a23-11ea-9bd5-641a1dd68908.png)

Reply:
Good morning @raphaeltimbo , and thanks for the tip. We will make the modifications in the next commit right away.